pkg/network/server.go

```diff
@@ -319,6 +321,7 @@ func (s *Server) Shutdown() {
 	}
 	close(s.quit)
 	<-s.relayFin
+	<-s.runFin
```

Comment:
We also need the same fix for `broadcastTxLoop` routine. It doesn't affect tests, but for proper node functioning we'd better ensure that this loop is ended normally.

Reply:
what about txHandlerLoop. should it have the same fix? like we still have the race and the only thing left is txHandlerLoop. or maybe transports

Reply:
Let's fix it as far, but that's the case when it should be done with wait group, because there are multiple tx handler loops started depending on the server settings.

> or maybe transports
We don't need it for transports since they are properly shutted down in `(s *Server) Shutdown()`.

Reply:
> like we still have the race and the only thing left is txHandlerLoop

`TestTryInitStateSync/module_already_initialized ` failing not because of this. It's failing because of inaccurate test structure, let's discuss under https://github.com/nspcc-dev/neo-go/pull/3307#discussion_r1492445285.